core/src/test/java/org/apache/iceberg/TestSnapshotManager.java

```diff
@@ -250,4 +248,305 @@ public void testCherryPickOverwrite() {
         lastSnapshotId, table.currentSnapshot().snapshotId());
     validateTableFiles(table, FILE_A, FILE_B);
   }
+
+  @Test
+  public void testCreateBranch() {
+    table.newAppend()
+        .appendFile(FILE_A)
+        .commit();
+    long snapshotId = table.currentSnapshot().snapshotId();
+    // Test a basic case of creating a branch
+    table.manageSnapshots()
+        .createBranch("branch1", snapshotId)
+        .commit();
+    SnapshotRef expectedBranch = table.ops().refresh().ref("branch1");
+    Assert.assertTrue(expectedBranch != null &&
+            expectedBranch.equals(SnapshotRef.branchBuilder(snapshotId).build()));
+
+    // Trying to create a branch with an existing name should fail
+    AssertHelpers.assertThrows("Should fail validation check if we try and create branch again",
```

Comment:
Style: personal pronouns ("I" and "we") should be avoided in comments, error messages, and assertion context.